backend/src/v5/models/metadata.js

```diff
@@ -33,25 +33,25 @@ Metadata.getMetadataByQuery = async (teamspace, model, query, projection) => {
 };
 
 Metadata.updateCustomMetadata = async (teamspace, model, metadataId, changeSet) => {
-	const { metadata } = await Metadata.getMetadataByQuery(teamspace, model, { _id: metadataId }, { metadata: 1 });
-	const metadataKeyLookup = metadata.reduce((a, b) => ({ ...a, [b.key]: 1 }), {});
-
-	changeSet.forEach((um) => {
-		const metadataExists = metadataKeyLookup[um.key];
-		if (metadataExists) {
-			const existingMetadata = metadata.find((m) => m.key === um.key);
-			if (um.value) {
-				existingMetadata.value = um.value;
-			} else {
-				const index = metadata.indexOf(existingMetadata);
-				metadata.splice(index, 1);
-			}
-		} else if (um.value) {
-			metadata.push({ ...um, custom: true });
+	const { metadata } = await Metadata.getMetadataById(teamspace, model, metadataId, { metadata: 1 });
+	const metadataKeyIndexLookup = metadata.reduce((parsedItems, currItem) => (
+		{
+			...parsedItems,
+			[currItem.key]: metadata.indexOf(currItem),
```

Comment:
@ChristosTsiotsias  you don't need to do `indexOf`, reduce should give you the current index if you look at the next parameter https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce#syntax

![image](https://user-images.githubusercontent.com/11945337/165918946-91b1960e-17d2-45fe-a1c1-2549b1988344.png)